csp_billing_adapter_microsoft/plugin.py

```diff
@@ -32,16 +32,15 @@
 
 log = logging.getLogger('CSPBillingAdapter')
 
-METADATA_ADDR = 'http://169.254.169.254/metadata'
-METADATA_API_VERSION = '?api-version=2021-12-13'
-METADATA_INSTANCE_PATH = '/instance'
-METADATA_TOKEN_PATH = '/identity/oauth2/token'
+METADATA_URL = 'http://169.254.169.254/metadata/'
+COMPUTE_URL = METADATA_URL + 'instance/compute?api-version={latest}'
+VERSIONS_URL = METADATA_URL + 'versions'
+METADATA_TOKEN = '/identity/oauth2/token?api-version={version}&resource=https://management.azure.com/'
 METADATA_TOKEN_RESOURCE = '&resource=https://management.azure.com/'
```

Comment:
seems like if you hard code the resource= in line 38 we do not need this anymore.

Reply:
yep, that was the intention